File: Astro/Models/UserPageContext.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Astro.Models
{
    public class UserPageContext
    {
        public int PublishedPosts { get; set; }
        public int AllLikes { get; set; }
        public int AllReports { get; set; }
        public CameraInfo CameraInfo { get; set; }
        public List<ChartInfo> ChartInfo { get; set; }
    }
}

File: Astro/ClientApp/src/app/models/UserPageContext.ts
export class CameraInfo {
    id: number;
    camera: string = '';
    cameraLens: string = '';
}

export class ChartInfo {
    name: string;
    value: number
}

export class UserPageContext {
    publishedPosts: number;
    allLikes: number;
    allReports: number;
    cameraInfo: CameraInfo = new CameraInfo();
    chartInfo: ChartInfo[];
}